lib/src/chunking.rs

```diff
@@ -364,78 +380,447 @@ impl Chunking {
     }
 }
 
-type ChunkedComponents<'a> = Vec<&'a ObjectSourceMetaSized>;
-
+#[cfg(test)]
 fn components_size(components: &[&ObjectSourceMetaSized]) -> u64 {
     components.iter().map(|k| k.size).sum()
 }
 
 /// Compute the total size of a packing
 #[cfg(test)]
-fn packing_size(packing: &[ChunkedComponents]) -> u64 {
+fn packing_size(packing: &[Vec<&ObjectSourceMetaSized>]) -> u64 {
     packing.iter().map(|v| components_size(v)).sum()
 }
 
-fn sort_packing(packing: &mut [ChunkedComponents]) {
-    packing.sort_by(|a, b| {
-        let a: u64 = components_size(a);
-        let b: u64 = components_size(b);
-        b.cmp(&a)
+fn mean(data: &[u64]) -> Option<f64> {
+    let sum = data.iter().sum::<u64>() as f64;
+    let count = data.len();
+    match count {
+        positive if positive > 0 => Some(sum / count as f64),
+        _ => None,
+    }
+}
+
+fn std_deviation(data: &[u64]) -> Option<f64> {
+    match (mean(data), data.len()) {
+        (Some(data_mean), count) if count > 0 => {
+            let variance = data
+                .iter()
+                .map(|value| {
+                    let diff = data_mean - (*value as f64);
+                    diff * diff
+                })
+                .sum::<f64>()
+                / count as f64;
+            Some(variance.sqrt())
+        }
+        _ => None,
+    }
+}
+
+fn median_absolute_deviation(data: &mut Vec<u64>) -> (f64, f64) {
+    //Sort data
```

Comment:
Why the commented out code here?

Reply:
Because the give data is already sorted in descending order. Sorting a sorted array could be computationally expensive and redundant. 
This leads to the question whether functions should be independent of any assumptions of the given input at the cost of efficiency